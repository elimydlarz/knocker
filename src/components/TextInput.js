import React, { Component } from 'react';
import '../css/TextInput.css';

export default class TextInput extends Component {
  render() {
    return (
      <div className='TextInput'>
        <input type="text" name={this.props.name}
          value={this.props.value}
          placeholder={this.props.label}
          onChange={this.props.changeHandler} />
      </div>
    );
  }
}
