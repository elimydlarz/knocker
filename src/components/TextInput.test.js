import React from 'react';
import renderer from 'react-test-renderer';
import TextInput from './TextInput';

describe('TextInput', () => {
  it('does not explode', () => {
    const component = renderer.create(
      <TextInput
        name='COOL NAME'
        label='COOL FIELD'
        value='COOL VALUE'
        changeHandler={() => {}} />
    )

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
