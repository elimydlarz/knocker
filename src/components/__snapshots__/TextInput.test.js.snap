exports[`TextInput does not explode 1`] = `
<div
  className="TextInput">
  <input
    name="COOL NAME"
    onChange={[Function changeHandler]}
    placeholder="COOL FIELD"
    type="text"
    value="COOL VALUE" />
</div>
`;
